backend/curriculum_tracking/management/commands/import_curriculum.py

```diff
@@ -0,0 +1,78 @@
+from core.models import Curriculum
+from curriculum_tracking.models import (
+    CurriculumContentRequirement,
+    ContentItemOrder,
+    ContentItem,
+)
+from curriculum_tracking.card_generation_helpers import get_ordered_content_items
+import json
+from django.core.management.base import BaseCommand
+import pandas as pd
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        json_file = "dev_helpers/data/intro-to-tilde-course.json"
+        save_curriculum_to_db(json_file)
+
+
+def save_curriculum_to_db(json_file):
+    with open(json_file) as f:
+        data = json.load(f)
+
+    # keys in dictionary `data` (4) `content_item_orders`, `content_items`, `curriculum`, `curriculum_content_requirements`
+    create_content_items(data['content_items'])
+    create_content_item_orders(data['content_item_orders'])
+    create_curriculum(data['curriculum'])
+    create_curriculum_content_requirements(data['curriculum_content_requirements'])
+
+def find_url_index(link, model):
```

Comment:
This function can be better. 

This is the same:

```
def find_url_index(link, model):
    if link == None:
        return 

    for index, value in enumerate(model.objects.all()):
        if value.url == link:
            return model.objects.filter(url=link).first()
```

You don't use `index`, so this is better:

```
def find_url_index(link, model):
    if link == None:
        return 

    for  value in model.objects.all():
        if value.url == link:
            return model.objects.filter(url=link).first()
```

Now let's look at that loop... first you fetch every single content item that exists in the database, then you fetch the one that you just matched

```
    for  value in model.objects.all():
        if value.url == link
            also_the_value =  model.objects.filter(url=link).first()
            assert value == also_the_value, "this should never raise an exception, because the sql filter you just performed was also performed by your loop"
            return also_the_value
```

A simpler version would be:

```
return model.objects.get(url=link) 
```

So basically:

```
def find_url_index(link, model):
    if link:
        return model.objects.get(url=link) 
```

This is the sameas your version. Except your version doesn't raise DoesNotExist errors. These errors would be useful because they make us aware of bugs

Also, you only ever call this with model = ContentItem. I dont think it would make sense to call it with any other model. So a further simplification is:

```
def find_url_index(link):
    if link:
        return ContentItem.objects.get(url=link) 
```

And lastly, good code is clear and easy to read. The function name does not describe what the function does.

how about 
```
def get_content_item_from_url(url):
    if url:   # why introduce a new name if we are already calling it url. Consistency is best
           return ContentItem.objects.get(url=url) 

```

Reply:
funny thing, initially I started with ContenItem and changed it later on to model